helix-term/src/commands/typed.rs

```diff
@@ -1801,22 +1800,48 @@ fn toggle_option(
     let pointer = format!("/{}", key.replace('.', "/"));
     let value = config.pointer_mut(&pointer).ok_or_else(key_error)?;
 
-    let Value::Bool(old_value) = *value else {
-        anyhow::bail!("Key `{}` is not toggle-able", key)
+    *value = match value.as_bool() {
+        Some(value) => serde_json::Value::Bool(!value),
+        None => {
+            if args.len() != 2 {
+                anyhow::bail!(
+                    "Bad arguments. For non-boolean configurations use: `:toggle key val1,val2`"
+                );
+            }
+            if !value.is_string() {
+                anyhow::bail!("Bad configuration. Cannot cycle non-string configurations");
+            }
+
+            // The comma separated values to cycle
+            let first_arg = &args[1];
+            let field_error = |_| anyhow::anyhow!("Could not parse field `{}`", first_arg);
+
+            first_arg
+                .split_once(',')
+                .map(|(first, remaining)| {
+                    Ok(serde_json::Value::String(
+                        [first]
+                            .into_iter()
+                            .chain(remaining.split(','))
+                            .skip_while(|e| e != value)
+                            .nth(1)
+                            .unwrap_or(first)
+                            .to_string(),
+                    ))
+                })
+                .unwrap_or_else(|| first_arg.parse().map_err(field_error))?
+        }
     };
 
-    let new_value = !old_value;
-    *value = Value::Bool(new_value);
-    // This unwrap should never fail because we only replace one boolean value
-    // with another, maintaining a valid json config
-    let config = serde_json::from_value(config).unwrap();
+    let status = format!("Option `{}` is now set to `{}`", key, value);
```

Comment:
```suggestion
    let status = format!("'{key}' is now set to '{value}'");
```

Reply:
Done in https://github.com/helix-editor/helix/pull/4411/commits/998426efcf754b738914a9d92b04e1a6a5bca5aa